docs/en_US/config/dns.md

```diff
@@ -0,0 +1,123 @@
+
+ * When the value is `"localhost"`, it means the machine's preset DNS configuration is used.
+
+ * When the value is in the form of `"https://host:port/dns-query"`, such as `"https://dns.google/dns-query"`, V2Ray will use `DNS over HTTPS` (RFC8484, short for  DOH) to query.  Some service providers have certificates for IP aliases and can write IP directly, such as `https://1.1.1.1/dns-query`.  You can also use non-standard ports and paths, such as `"https://a.b.c.d:8443/my-dns-query"` (4.22.0+)
+
+ * When the value is in the form of `"https+local://host:port/dns-query"`, such as `"https+local://dns.google/dns-query"`, V2Ray will use `DOH local mode`  For query, that is, DOH requests will not go through Routing/Outbound and other components, but directly request externally to reduce time-consuming.  Generally suitable for use on the server side.  Non-standard ports and paths can also be used. (4.22.0+)
+
+:::tip
+When using `localhost`, the DNS request of this machine is not controlled by V2Ray, and additional configuration is needed to make DNS request forwarded by V2Ray.
+
+The DNS clients initialized by different rules will be reflected in the V2Ray startup log with the ʻinfo` level, such as `local DOH`, `remote DOH` and ʻudp` etc. modes.  (4.22.0+)
+:::
+
+> `clientIp`: string
+
+The IP address of the current system is used to inform the server of the location of the client during DNS query.  It cannot be a private address.
+
+> `tag`: string
+
+(V2Ray 4.13+) The query traffic sent by this DNS, except for `localhost` and `DOHL_` modes, will carry this identifier, which can be matched with ʻinboundTag` in the routing.
+
+## ServerObject
+
+```json
+{
+    "address": "1.2.3.4",
+    "port": 5353,
+    "domains": [
+        "domain:v2ray.com"
+    ],
+    "expectIPs": [
+        "geoip:cn"
+    ]
+}
+```
+
+> `Address`: address
+
+ DNS server address, such as `"8.8.8.8"`.  For ordinary DNS IP addresses that only support UDP protocol DNS servers, if the address is in the form of a URL beginning with `"https://"` or `"https+local://"`, DOH mode is used, and the rules are the same as strings  DOH configuration of the mode.
+
+ > `port`: number
+
+ DNS server port, such as `53`.  This item defaults to `53` by default.  When using DOH mode, this item is invalid, non-standard port should be specified in URL.
+
+ > `domains`: \[string\]
+
+ A list of domain names. The domain names contained in this list will be queried by this server first.  The domain name format is the same as in [Routing Configuration](routing.md#ruleobject).
+
+ > `ExpectIPs`:\[string\]
+
+ (V2Ray 4.22.0+) An IP range list, the format is the same as in [Routing Configuration](routing.md#ruleobject).
+
+ When this option is configured, V2Ray DNS will verify the returned IP and only return the addresses in the expectIPs list.
+
+ If this item is not configured, the IP address will be returned as it is.
```

Comment:
```suggestion
If this item is not configured, the origin IP address will be returned.
```

Reply:
`ExpectIPs` is not a item for rewrite returned data, so i will insisted in my translation, but i made a supplement for here.